Preview and Transfer (#545)
hellohaptik/docs_haptik: docs/bot-builder/basic/transferbot.md

- The Business Manager should be transferred to production before the bot.

## Transferring your Business from Staging to Production

The Business Manager is responsible for the configuration of multiple key characteristics of the bot such as the appearance of the bot, bot break messages, deployment settings, bot prompts, and task boxes. Hence, the business needs to be transferred to the production environment.

You can click on the **Transfer button** as shown in the below image.

![TB_1](assets/Business14.png)

> The Business should be transferred again if there are modifications made after the Business is live.

## Transferring your Bot from Staging to Production

![image](https://user-images.githubusercontent.com/75118325/124393008-03573d00-dd16-11eb-933b-6875925b5c16.png)

Once the bot is built and has passed the functional testing, you can transfer the bot to production. 

Click on the **Preview & Transfer** button. You will see a modal that generates a preview of changes as seen below - 

![preview1](https://user-images.githubusercontent.com/75118325/124290857-91de8980-db71-11eb-96b2-dbee73b467f3.gif)

The modal will display the changes that were made to the bot.

> ![image](https://user-images.githubusercontent.com/75118325/124392874-5a104700-dd15-11eb-9f8a-435d0c690e77.png)

Reviewer: Will have to change this image. It contains the grey icon

* Indicators

#### Changes

You will see the preview for 3 sections of the bot where the changes were done - 
1. Entities
2. Skill Set
3. General Nodes

![image](https://user-images.githubusercontent.com/75118325/124419235-bb1e3600-dd7a-11eb-91ed-2f4b67ba94f6.png)

#### Contributors

In the **Contributors** section, you will see all the users who have made changes to the bot.

![image](https://user-images.githubusercontent.com/75118325/124393292-64cbdb80-dd17-11eb-84e4-7b545a39d1e3.png)

#### Indicators

There are different indicators to denote different types of changes made on the bot. 

> When you hover over the indicators in the **Change** section, you will be able to check the actions taken by the **Contributor** along with the timestamp.

When a user had added a new element such as a new entity, new Skill Set, and so on, the indicator in the **Changes** section will be **_green_**.

When some element had been deleted such as deactivating a node, removing, and so on, the indicator in the **Changes** section will be **_red_**.

When there are small edits made to any element such as entity, the bot says, the indicator in the **Changes** section will be **_yellow_**.

> When there are any changes made to a Skill, such as changing the bot says, adding a new node, adding a new connection and so on, the indicator in the **Changes** section will be **_gray_**.

Reviewer: ```suggestion
```

> ![image](https://user-images.githubusercontent.com/75118325/124393354-b96f5680-dd17-11eb-96e1-a7390b497908.png)

Reviewer: Screenshot will have to be changed

### Changes to Entities

You can check when a user has made any changes to the entity, for example, changing the entity settings or changing the entity reprompt message, and so on.

When there are edits made to the already existing entity, the indicator will be **_yellow_**.

> ![image](https://user-images.githubusercontent.com/75118325/124393523-8a0d1980-dd18-11eb-872e-929d741d3209.png)

Reviewer: screenshot to be changed

When a user is creating and adding a new entity, the indicator will be **_green_**. 

> ![image](https://user-images.githubusercontent.com/75118325/124393566-b5900400-dd18-11eb-82d5-8417bbae11df.png)

Reviewer: screenshot to be changed

### Changes to Skill Sets

The **_gray_** indicator present along Skill Set denotes there were changes made in the particular Skill Set.

> ![image](https://user-images.githubusercontent.com/75118325/124417674-4c8ba900-dd77-11eb-8bc5-7951bb513533.png)

Reviewer: Remove this entire section

In the Skill Set section, you will be able to see the changes made to - 
* Nodes
  * You can check the new nodes created, indicated as **_green_**.
  * If a node was deactivated, it will be indicated as **_yellow_**.
  >
  > ![image](https://user-images.githubusercontent.com/75118325/124417966-f9662600-dd77-11eb-9558-10aedc45929c.png)
  > 

* Connections
  * A new connection created between nodes is indicated as **_green_**.
  * If a connection is deleted between nodes, it is indicated as **_red_**.
  > 
  > ![image](https://user-images.githubusercontent.com/75118325/124418127-65488e80-dd78-11eb-9445-14a13466c635.png)
  > 

When you hover over the indicator, you will be able to see the **Contributors** name. If there are multiple Contributors who made changes to a particular Skill Set, you can check their names and the changes they made to the particular Skill Set such as creating a new node, creating a new connection, etc.

> ![image](https://user-images.githubusercontent.com/75118325/124418385-ffa8d200-dd78-11eb-93ca-0a346a590cae.png)

Reviewer: screenshot to be changed.

  > 

When you hover over the indicator, you will be able to see the **Contributors** name. If there are multiple Contributors who made changes to a particular Skill Set, you can check their names and the changes they made to the particular Skill Set such as creating a new node, creating a new connection, etc.

![image](https://user-images.githubusercontent.com/75118325/124418385-ffa8d200-dd78-11eb-93ca-0a346a590cae.png)

### Changes to General Nodes

For all the newly created nodes in the General Nodes section, the indicator will be **_green_**.

When a contributor has edited an already existing node, the indicator will be **_yellow_**.

![image](https://user-images.githubusercontent.com/75118325/124418491-3383f780-dd79-11eb-9fec-1c4dbadd1b09.png)

<hr>

When someone is trying to transfer an empty bot i.e. a bot where Skill Set and Skill is not created, you will see a pop-up as shown below - 

![image](https://user-images.githubusercontent.com/75118325/124418722-bf961f00-dd79-11eb-8514-ce0ef872e040.png)

When there are no changes made to the bot since the last transfer, you will see a pop-up as shown below - 

> ![image](https://user-images.githubusercontent.com/75118325/124419709-aee6a880-dd7b-11eb-833f-ab61c5d80043.png)

Reviewer: For both scenarios, the image remains the same. The _refigerator_ one

> When a **Contributor** has saved new changes on the bot at the time when you initiated the transfer, you will get a pop-up to refresh the preview.

Reviewer: ```suggestion
When another **Contributor** has saved new changes on the bot at the same time when you initiated the transfer, you will get a pop-up indicating the same. You can refresh the preview to see the new changes or go ahead with the transfer.
```

![preview and transfer 1](https://user-images.githubusercontent.com/75118325/126495483-c9179f8c-60ac-42b5-a184-99a8e75523e2.gif)

The modal will display the changes that were made to the bot.

The modal has 3 sections - 
* Changes
* Contributors
* Indicators

#### Changes

You will see the preview for 3 sections of the bot where the changes were done - 
1. Entities
2. Skill Set
3. FAQs

![image](https://user-images.githubusercontent.com/75118325/126495622-757a8cf2-5a7a-4a3c-919d-966bc9a9921a.png)

#### Contributors

In the **Contributors** section, you will see all the users who have made changes to the bot.

![image](https://user-images.githubusercontent.com/75118325/126495762-80586741-3aac-4cca-86ed-4a972a8b7864.png)

#### Indicators

There are different indicators to denote different types of changes made on the bot. 

> When you hover over the indicators in the **Change** section, you will be able to check the actions taken by the **Contributor** along with the timestamp.

When a user had added a new element such as a new entity, new Skill Set, and so on, the indicator in the **Changes** section will be **_green_**.

> When some element had been deleted such as deactivating a node, removing, and so on, the indicator in the **Changes** section will be **_red_**.

Reviewer: ```suggestion
When some element had been deleted such as deleting a connection, the indicator in the **Changes** section will be **_red_**.
```

> When there are small edits made to any element such as entity, the bot says, the indicator in the **Changes** section will be **_yellow_**.

Reviewer: ```suggestion
When there are small edits made to any element such as the bot says of a node, the indicator in the **Changes** section for that particular element will be **_yellow_**.
```
